docs/articles/semantic_search_news.md

```diff
@@ -0,0 +1,330 @@
+
+We are embedding
+
+headlines
+news body (short description)
+and date
+to be able to search for
+
+notable events, or
+related articles to a specific story.
+There is a possibility to skew the results towards older or fresher news, and also to influence the results using a specific search term.
+
+In [2], change `mimetype` to ‘colab’ in alt.renderers.enable
+
+## Boilerplate
+
+First, we'll install Superlinked.
+
+```python
+%pip install superlinked==9.43.0
+```
+
+And then do our imports and constants...
+
+(Note: Below, change `alt.renderers.enable(“mimetype”)` to `alt.renderers.enable('colab')` if you’re running this in google colab. Keep “mimetype” if you’re executing in github.)
+
+```python
+from datetime import datetime, timedelta, timezone
+
+import os
+import sys
+import altair as alt
+import pandas as pd
+
+from superlinked.evaluation.charts.recency_plotter import RecencyPlotter
+from superlinked.framework.common.dag.context import CONTEXT_COMMON, CONTEXT_COMMON_NOW
+from superlinked.framework.common.dag.period_time import PeriodTime
+from superlinked.framework.common.schema.schema import Schema
+from superlinked.framework.common.schema.schema_object import String, Timestamp
+from superlinked.framework.common.schema.id_schema_object import IdField
+from superlinked.framework.common.parser.dataframe_parser import DataFrameParser
+from superlinked.framework.dsl.executor.in_memory.in_memory_executor import (
+    InMemoryExecutor,
+    InMemoryApp,
+)
+from superlinked.framework.dsl.index.index import Index
+from superlinked.framework.dsl.query.param import Param
+from superlinked.framework.dsl.query.query import Query
+from superlinked.framework.dsl.query.result import Result
+from superlinked.framework.dsl.source.in_memory_source import InMemorySource
+from superlinked.framework.dsl.space.text_similarity_space import TextSimilaritySpace
+from superlinked.framework.dsl.space.recency_space import RecencySpace
+
+alt.renderers.enable("mimetype") # NOTE: to render altair plots in colab, change 'mimetype' to 'colab'
+alt.data_transformers.disable_max_rows()
+alt.data_transformers.disable_max_rows()
+pd.set_option("display.max_colwidth", 190)
+```
+
+```python
+YEAR_IN_DAYS = 365
+TOP_N = 10
+DATASET_URL = "https://storage.googleapis.com/superlinked-notebook-news-dataset/business_news.json"
+# as the dataset contains articles from 2022 and before, we can set our application's "NOW" to this date
+END_OF_2022_TS = int(datetime(2022, 12, 31, 23, 59).timestamp())
+EXECUTOR_DATA = {CONTEXT_COMMON: {CONTEXT_COMMON_NOW: END_OF_2022_TS}}
+```
+
+### Prepare & explore dataset
+
+```python
+NROWS = int(os.getenv("NOTEBOOK_TEST_ROW_LIMIT", str(sys.maxsize)))
+business_news = pd.read_json(DATASET_URL, convert_dates=True).head(NROWS)
+```
+
+```python
+# we are going to need an id column
+business_news = business_news.reset_index().rename(columns={"index": "id"})
+# we need to handle the timestamp being set in milliseconds
+business_news["date"] = [
```

Comment:
@morkapronczay
meaning the original timestamp is in milliseconds? in [kaggle](https://www.kaggle.com/datasets/rmisra/news-category-dataset/data), it looks like it's just dates - eg "date":string"2022-09-23"

Reply:
not sure about the kaggle text. Might have been me altering it when I filtered the dataset for myself. (probably when I read it it automatically converted it to date and I wrote it out like this)

Reply:
@morkapronczay
The kaggle is just dates. 
How should we discuss the conversion to seconds from milliseconds? 
Or does the following snippet take care of converting from milliseconds (to seconds) whether your dataset has been read into milliseconds or not:

```python
# we need to handle the timestamp being set in milliseconds
business_news["date"] = [
    int(date.replace(tzinfo=timezone.utc).timestamp()) for date in business_news.date
]
```

Reply:
@morkapronczay no rush on this question above, just pinging you for when you get time. Thanks!